src/module_specifier.rs

```diff
@@ -54,13 +55,27 @@ fn specifier_from_path(_path: PathBuf) -> ModuleSpecifier {
 pub fn resolve_import(
   specifier: &str,
   referrer: &ModuleSpecifier,
+  maybe_npm_resolver: Option<&dyn NpmResolver>,
 ) -> Result<ModuleSpecifier, SpecifierError> {
   let url = match ModuleSpecifier::parse(specifier) {
     // 1. Apply the URL parser to specifier.
-    //    If the result is not failure, return he result.
+    //    If the result is not failure, return the result.
     Ok(url) => url,
 
-    // 2. If specifier does not start with the character U+002F SOLIDUS (/),
+    // 2. If specifier is a node specifier without node: scheme, return
+    //    the specifier with node: scheme added.
+    Err(ParseError::RelativeUrlWithoutBase)
+      if maybe_npm_resolver.map_or(false, |npm_resolver| {
+        npm_resolver.is_builtin_node_module_name(specifier)
+      }) =>
+    {
```

Comment:
I think we should move this check to a higher level in:

https://github.com/denoland/deno_graph/blob/aa9236c40a2b8e2514c1f08e399b2c0ec70f81fd/src/graph.rs#L1551-L1562

https://github.com/denoland/deno_graph/pull/296 will make this easier.

That way this behaviour isn't dependent on an implentor of the `Resolver` calling this `resolve_import` function and needing access to the npm resolver. Basically, if a user provides an `NpmResolver` to deno_graph then they'll get this behaviour.

Reply:
Moved the check of bare builtin node specifier. Now it is checked after `resolver.resolve` is applied.